src/components/radio/radio.ts

```diff
@@ -84,16 +85,14 @@ export class MdRadioGroup implements AfterContentInit, ControlValueAccessor {
   @ContentChildren(forwardRef(() => MdRadioButton))
   private _radios: QueryList<MdRadioButton> = null;
 
+  constructor(private _changeDetectionRef: ChangeDetectorRef) {}
+
   /**
    * Initialize properties once content children are available.
    * This allows us to propagate relevant attributes to associated buttons.
    */
   ngAfterContentInit() {
-    if (this._name == null) {
-      this.name = `md-radio-group-${_uniqueIdCounter++}`;
-    } else {
-      this._updateChildRadioNames();
-    }
+    this._updateChildRadioNames();
```

Comment:
As discussed, change this such that the child radio pulls its name from the parent instead of the other way around.